### DISCUSSION TITLE
JK-B2A24S-15P

### DISCUSSION THREAD

**Discussion by alikhan968** [syssi/esphome-jk-bms#620] [Q&A]
Hi,
I simply want to read UART data with ESP32 using GPS PORT.
I need help to find some code that already available to decode the protocol of JK BMS.
I am receiving following data on serial by simply reading UART.
`11:31:33.212 -> DE  11:31:33.212 -> 6B  11:31:33.212 -> EF  11:31:33.212 -> 63  11:31:33.212 -> EF  11:31:33.212 -> 43  11:31:33.212 -> 60  11:31:33.212 -> 92  11:31:33.212 -> 67  11:31:33.212 -> 43  11:31:33.212 -> 32  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 71  11:31:33.212 -> C5  11:31:33.212 -> 84  11:31:33.212 -> 43  11:31:33.212 -> 42  11:31:33.212 -> 41  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 86  11:31:33.212 -> 42  11:31:33.212 -> 42  11:31:33.212 -> 40  11:31:33.212 -> C2  11:31:33.212 -> 63  11:31:33.212 -> EF  11:31:33.212 -> 7B  11:31:33.212 -> EF  11:31:33.212 -> 23  11:31:33.212 -> FF  11:31:33.797 -> 77  11:31:33.797 -> BF  11:31:33.797 -> F5 `

Any hint would be highly appreciated.
Thanks

**Comment by syssi** [comment]
If you flash this YAML configuration to your ESP you will be able to decode the GPS port traffic:

https://github.com/syssi/esphome-jk-bms/blob/main/esp32-example.yaml
